### PR TITLE
[CU-2j8gc2a] MVP of `rebalance` without `get_apy`

### PR DESCRIPTION
**Pull request by saruman9**
## Issue

[CU-2j8gc2a](https://app.clickup.com/t/2j8gc2a)

## Description

First attempt to solve the task. Only for review and consulting.

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2j8gc2a Story 1 (MVP of rebalance without `get_apy`)](https://app.clickup.com/t/2j8gc2a)

**Review by kevin-composable — APPROVED** [review]

File: frame/pablo/src/lib.rs [hunk 1]
```diff
@@ -670,6 +669,10 @@ pub mod pallet {
 		type AccountId = T::AccountId;
 		type PoolId = T::PoolId;
 
+		fn pools() -> Vec<Self::PoolId> {
```
Comment (belousm) [inline frame/pablo/src/lib.rs:672]: Isn't it excess memory allocation? I mean creation of `Vec` with all pools. Is't better to iterate over storage in `get_pool()` function and use Closure there? Or it's the same stuff?!

Comment (belousm) [inline frame/pablo/src/lib.rs:672]: Isn't it excess memory allocation? I mean creation of `Vec` with all pools. Is't better to iterate over storage in `get_pool()` function and use Closure there? Or it's the dame stuff?!

**Review by kevin-composable — APPROVED** [review]

**Review by kevin-composable — APPROVED** [review]

**Comment by saruman9** [comment]
See #1130.

**Pull request closed without merging**